File: jasmine/src/thermostat.js
function Thermostat() {
  this.temp = 20
  this.MinTemp = 10
  this.MaxTemp = 25
  this.PSM = true
};

Thermostat.prototype.increase = function() {
  if(this.temp === this.MaxTemp){
    throw new TypeError ("You reached the maximum temprature!");
  };
    this.temp += 1
};

Thermostat.prototype.decrease = function() {
  if(this.temp === this.MinTemp){
    throw new TypeError ("You reached the minimum temprature!");
  };
    this.temp -= 1
  };

Thermostat.prototype.switchOff = function() {
  this.PSM = false
  this.MaxTemp = 32
};

Thermostat.prototype.switchOn = function() {
  this.PSM = true
  this.MaxTemp = 25
};

Thermostat.prototype.reset = function() {
  this.temp = 20
};

Thermostat.prototype.usage = function() {
    if(this.temp < 18){
    return "low-usage"
  } else if(this.temp < 25){
    return "medium-usage"
  } else {
    return "high-usage"
  };
};

File: pre-exercises/python/thermostat.py
class Thermostat:
    def __init__(self):
        self.temp = 20
        self.MinTemp = 10
        self.MaxTemp = 25
        self.PSM = True
        self.status = "ON"

    def increase(self):
        if self.temp >= self.MaxTemp:
            raise Exception("You reached the maximum temprature!")
        self.temp += 1
    def decrease(self):
        if self.temp == self.MinTemp:
            raise Exception("You reached the minimum temprature!")
        self.temp -= 1

    def switchOn(self):
        self.PSM = True
        self.MaxTemp = 25
        self.status = "ON"
    def switchOff(self):
        self.PSM = False
        self.MaxTemp = 32
        self.status = "OFF"

    def reset(self):
        self.temp = 20
    def usage(self):
        if self.temp < 18:
            return "low-usage"
        elif self.temp < 25:
            return "medium-usage"
        else:
            return "high-usage"
            
